src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -256,35 +359,45 @@ var (
 	replicaReOnce sync.Once
 )
 
-// configureReplicas parses the partition flag with format; foo/0:1-0,2-1,3-2 or 0:1,2,3
-// It extracts letters after the colon and return as adminapi.Replicas.
-func configureReplicas(partition string, currentReplicas rpadmin.Replicas) (rpadmin.Replicas, error) {
+// extractReplicaChanges parses the node-core from the partition flag with format:
+// foo/0:1-0,2-1,3-2 or 0:1,2,3. It returns the full rpadmin.Replicas changes,
+// and a subset of only the core changes requested.
+func extractReplicaChanges(partitionFlag string, currentReplicas rpadmin.Replicas) (allReplicas, coreReplicas rpadmin.Replicas, err error) {
 	replicaReOnce.Do(func() {
+		// This regexp captures the sequence of numbers (nodes) after a colon.
+		// The sequence can be a single number, two numbers separated by a
+		// hyphen (node-core), or multiple of such numbers separated by commas.
 		replicaRe = regexp.MustCompile(`^[^:]+:(\d+(?:-\d+)?(?:,\d+(?:-\d+)?)*)$`)
 	})
-	m := replicaRe.FindStringSubmatch(partition)
+	// Match[0]: Full Match.
+	// Match[1]: String after the colon. (node[-core],...)
+	m := replicaRe.FindStringSubmatch(partitionFlag)
 	if len(m) == 0 {
-		return nil, fmt.Errorf("invalid format for %s", partition)
+		return nil, nil, fmt.Errorf("invalid format for %s; check --help text", partitionFlag)
 	}
-	var newReplicas rpadmin.Replicas
 	for _, nodeCore := range strings.Split(m[1], ",") {
-		if split := strings.Split(nodeCore, "-"); len(split) == 1 {
+		if split := strings.Split(nodeCore, "-"); len(split) == 1 { // Node change: -p 0:1,2,3
 			node, _ := strconv.Atoi(split[0])
 			core := findCore(node, currentReplicas)
-			newReplicas = append(newReplicas, rpadmin.Replica{
-				NodeID: node,
-				Core:   core,
-			})
-		} else {
+			allReplicas = append(allReplicas, rpadmin.Replica{NodeID: node, Core: core})
+		} else { // Node/Core changes: -p 0:1-0,2-3,3
 			node, _ := strconv.Atoi(split[0])
-			core, _ := strconv.Atoi(split[1])
-			newReplicas = append(newReplicas, rpadmin.Replica{
-				NodeID: node,
-				Core:   core,
-			})
+			currentCore := findCore(node, currentReplicas)
+			if currentCore == -1 { // -1 == not found == new node
+				return nil, nil, fmt.Errorf("node '%v': this command does not support updating cores for replicas on new nodes", node)
+			}
+			newCore, _ := strconv.Atoi(split[1])
+			allReplicas = append(allReplicas, rpadmin.Replica{NodeID: node, Core: newCore})
+			// We only append core changes. If the requested core is the same, we don't bother.
+			if currentCore != newCore {
+				coreReplicas = append(coreReplicas, rpadmin.Replica{NodeID: node, Core: newCore})
+			}
 		}
 	}
-	return newReplicas, nil
+	if len(allReplicas) != len(currentReplicas) {
+		return nil, nil, fmt.Errorf("cannot configure %s; cannot modify replication factor, current replica count: %v; replica count requested: %v", partitionFlag, len(currentReplicas), len(allReplicas))
```

Comment:
Can we guide users the alternative command? `rpk topic alter-config TOPIC -s replication.factor=x`

Reply:
Good one, I've added this to the error. I'll push the changes once we have confirmation in the other comment thread. :+1: